### DISCUSSION TITLE
Type `Reference<Entity>` is not assignable to type '{ id: number; } & Reference<Entity>'

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#3156] [Q&A]
In `mikro-orm@5.1.2`, this worked, but it does not work in `mikro-orm@5.1.4`.

Basically, I reference an entity (`RefEntity`; it contains `id` property) in another entity (`Entity`).

```typescript
@Entity({schema: 'md', tableName: 'entity'})
export class Entity {
  @PrimaryKey()
  id!: number

  @ManyToOne(() => RefEntity, {wrappedReference: true, fieldName: 'ref_id', default: 1})
  ref: IdentifiedReference<RefEntity>

  constructor(dto: {ref: RefEntity}) {
    this.ref = Reference.create(dto.refId)
  }
}
```

However, now I get the following error:

```typescript
Type 'Reference<MonitorView>' is not assignable to type '{ id: number; } & Reference<MonitorView>'.
  Property 'id' is missing in type 'Reference<MonitorView>' but required in type '{ id: number; }'.

(property) Screen.view: {
    id: number;
} & Reference<MonitorView>
```

I noticed that there was a change related to references (see #3092), but I have no idea if this also changed the syntax.

Could you help me out please?

**Comment by B4nan** [comment]
Can you provide exact entity definition, including the target entity? That change you linked is not related, it does not affect any types. 

Also ensure you dont have multiple orm packages installed (eg core package twice), that could lead to incompatible types too. Wipe lock file and node modules and reinstall.

**Reply by tukusejssirs** [reply]
I have no idea what the exact problem was, but it was fixed by some other actions I did.

On the other hand, I tried to remove `package-lock.json` and `node_packages/` and reinstalled the packages, but it didn’t work all by itself, but I am sure that only a single version per package was installed. As for version of different `@mikro-orm/*` packages, all packages are installed and updated at once using `@latest` version.
